torch/csrc/jit/pickler.cpp

```diff
@@ -201,8 +201,12 @@ void Pickler::pushIValueImpl(const IValue& ivalue) {
 }
 
 void Pickler::pushIValue(const IValue& ivalue) {
-  // Check if reference ivalue has been saved before
-  if (ivalue.isPtrType()) {
+  bool isMutableType = ivalue.isPtrType() && !ivalue.isString() && !ivalue.isTuple();
```

Comment:
Memoization of tuples will change the behavior of `is` on Tuples. I am not sure whether this is an allowed semantic change. It is probably safer not to do it.

```
import pickle

a = (3, 4)
b = (a, a)

saved = pickle.dumps(b)
b_ = pickle.loads(saved)

print(b_[0] is b_[1]) # True
```